src/core/gltf2importer/meshparser_utils.cpp

```diff
@@ -0,0 +1,381 @@
+#include "meshparser_utils_p.h"
+
+#include <Qt3DRender/QBuffer>
+#include <Qt3DRender/QGeometry>
+#include <Qt3DRender/QAttribute>
+#include <Qt3DRender/QGeometryRenderer>
+
+#include <array>
+
+#include "mikktspace/mikktspace.h"
+
+namespace {
+
+std::vector<Qt3DRender::QAttribute::VertexBaseType> validVertexBaseTypesForAttribute(const QString &attributeName)
+{
+    // Standard Attributes
+    if (attributeName == Qt3DRender::QAttribute::defaultPositionAttributeName())
+        return { Qt3DRender::QAttribute::Float };
+    if (attributeName == Qt3DRender::QAttribute::defaultNormalAttributeName())
+        return { Qt3DRender::QAttribute::Float };
+    if (attributeName == Qt3DRender::QAttribute::defaultTangentAttributeName())
+        return { Qt3DRender::QAttribute::Float };
+    if (attributeName == Qt3DRender::QAttribute::defaultTextureCoordinateAttributeName())
+        return { Qt3DRender::QAttribute::Float,
+                 Qt3DRender::QAttribute::UnsignedByte,
+                 Qt3DRender::QAttribute::UnsignedShort };
+    if (attributeName == Qt3DRender::QAttribute::defaultTextureCoordinate1AttributeName())
+        return { Qt3DRender::QAttribute::Float,
+                 Qt3DRender::QAttribute::UnsignedByte,
+                 Qt3DRender::QAttribute::UnsignedShort };
+    if (attributeName == Qt3DRender::QAttribute::defaultColorAttributeName())
+        return { Qt3DRender::QAttribute::Float,
+                 Qt3DRender::QAttribute::UnsignedByte,
+                 Qt3DRender::QAttribute::UnsignedShort };
+    if (attributeName == Qt3DRender::QAttribute::defaultJointIndicesAttributeName())
+        return { Qt3DRender::QAttribute::UnsignedByte,
+                 Qt3DRender::QAttribute::UnsignedShort };
+    if (attributeName == Qt3DRender::QAttribute::defaultJointWeightsAttributeName())
+        return { Qt3DRender::QAttribute::Float,
+                 Qt3DRender::QAttribute::UnsignedByte,
+                 Qt3DRender::QAttribute::UnsignedShort };
+
+    return { Qt3DRender::QAttribute::UnsignedByte,
+             Qt3DRender::QAttribute::UnsignedShort,
+             Qt3DRender::QAttribute::UnsignedInt };
+}
+
+std::vector<uint> validVertexSizesForAttribute(const QString &attributeName)
```

Comment:
I would prefer to leave it like this. I would need to take into account that -1 in the check after this, which makes it less clear. In most cases this std::vector is not going to perform any allocation anyway, no?

Reply:
Well std::vector is always going to perform an allocation. What it does though is probably preallocate a given capacity so that it doesn't have to reallocate every time you insert a new elements. So I guess in the end the question is more about allocation time vs if branch. Note that QVarLengthArray might be interesting in this case as you can specify the maximum amount it should contain but fill in less elements than that if you want to which I guess would allow to not do allocations and not have to perform checks.

Reply:
You are right about std::vector. I thought it has some small buffer optimization or something like that in common implementations as std::string has.

Reply:
I have moved to QVarLengthArray